OpenRA.Game/Graphics/ModelRenderer.cs

```diff
@@ -279,6 +283,10 @@ static float[] ExtractRotationVector(float[] mtx)
 
 		public void BeginFrame()
 		{
+			if (isInFrame)
+				throw new InvalidOperationException(
+					"BeginFrame has already been called. A new frame cannot be started until EndFrame has been called.");
+			isInFrame = true;
```

Comment:
Please add an empty line before this. I'm also personally not a big fan of the line break in the exception statement.